### PR TITLE
Refactor/nbitcoin protocol

### PR DESCRIPTION
**Pull request by Aprogiena**
code cleanup in NBitcoin.Protocol

### CODE REVIEW

File: src/NBitcoin/Protocol/BitcoinSerializablePayload.cs [hunk 1]
```diff
@@ -1,30 +1,22 @@
 ﻿namespace NBitcoin.Protocol
 {
-	public class BitcoinSerializablePayload<T> : Payload where T : IBitcoinSerializable, new()
-	{
-		public BitcoinSerializablePayload()
-		{
+    public class BitcoinSerializablePayload<T> : Payload where T : IBitcoinSerializable, new()
+    {
+        private T @object = new T();
```
Comment (bokobza) [inline src/NBitcoin/Protocol/BitcoinSerializablePayload.cs]: Can we just have a property that's initialized in the constructor?
Also, please let's have a different name, like "o to the b to the j etc".

File: src/NBitcoin/Protocol/InventoryVector.cs [hunk 2]
```diff
@@ -22,56 +16,48 @@ public enum InventoryType : uint
...
+            }
+            set
+            {
+                this.type = (uint)value;
+            }
+        }
 
-		public InventoryVector()
-		{
+        private uint256 hash = uint256.Zero;
```
Comment (bokobza) [inline src/NBitcoin/Protocol/InventoryVector.cs]: same here, can we have the property defined like we like (without the backing field)?

**Comment by bokobza** [comment]
@Aprogiena is there 0 functionality change here?
My only comment would be to make sure you remove backing fields wherever possible.

File: src/NBitcoin/Protocol/BitcoinSerializablePayload.cs (hunk 1 above)
Comment (Aprogiena) [reply src/NBitcoin/Protocol/BitcoinSerializablePayload.cs]: i don't like it either :-) but being inside these classes i was afraid of changing something too much, but it seems like this is safe, so i've changed it now, it did propagate to quite many files though :/

**Comment by Aprogiena** [comment]
this should be 99.99% equivalent code. the most different stuff i've done so far is to replace Task.Factory.StartNew with Task.Run

File: src/NBitcoin/Protocol/InventoryVector.cs (hunk 2 above)
Comment (dangershony) [reply src/NBitcoin/Protocol/InventoryVector.cs]: Yeah you can't change it, `ref` will work on the same copy that was passed to the method.
This allows to use the serializer as both read and write

**Pull request merged**